edb/ir/ast.py

```diff
@@ -886,6 +886,20 @@ def is_global(self) -> bool:
         return self.is_implicit_global is not None
 
 
+class InlinedParameter(ImmutableBase):
+
+    # Parameter is used for both function parameters and runtime parameters.
+    # If a function is inlined in compile_FunctionCall, this is used to
+    # specify the function's parameters.
+
+    # During function compilation, arguments are converted from names to either
+    # int for positional argument or str for named argument.
+    arg_key: int | str
+
+    required: bool
+    is_global: bool
```

Comment:
The more typical way to do something like this in our compiler would probably be to substitute in `irast.Set` objects that have the same `path_id` as as the arguments but have an `irast.RefExpr` as their `expr`, to indicate that they are bound somewhere.

They should then get picked up by their path id when doing the backend compilation, without needing to add an extra map.